kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala

```diff
@@ -69,6 +69,12 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
   private val timeoutChecker =
     ThreadUtils.newDaemonSingleThreadScheduledExecutor(s"$name-timeout-checker")
 
+  @volatile private var _limiter: Option[SessionLimiter] = None
+
+  protected def setSessionLimiter(limiter: SessionLimiter): Unit = {
```

Comment:
shall we add this only in the server？

Reply:
> shall we add this only in the server？

I'm also not sure if we need to add it in Engine too? Since the engine can be shared maybe we can also add limiter to the engine. What do you think?

Reply:
Yes, maybe. It won't be late when we have plans on such a thing for engines. So let's focus on the PR's scope itself

Reply:
I just set the limiter in KyuubiSessionManager, so do I need to make some changes here?

Reply:
move all the logic from common to server would be more clear

Reply:
> move all the logic from common to server would be more clear

OK, thanks. I'll do that.

Reply:
@yaooqinn I need to decrement limit after close session, in KyuubiSessionManager's closeSession method we can't seem to implement it well. Because the session will be removed from handleToSession after closeSession.

Reply:
> @yaooqinn I need to decrement limit after close session, in KyuubiSessionManager's closeSession method we can't seem to implement it well. Because the session will be removed from handleToSession after closeSession.

Sorry, please ignore this comment, I have moved it.